cfep-05.md

```diff
@@ -0,0 +1,136 @@
+
+<table>
+<tr><td> Title </td><td> Providing pre-release builds on the conda-forge anaconda channel </td>
+<tr><td> Status </td><td> Proposed </td></tr>
+<tr><td> Author(s) </td><td> Eric Dill &lt;thedizzle@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Sept 09, 2016</td></tr>
+<tr><td> Updated </td><td> Sept 09, 2016</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+There are a number of recipe maintainers who have asked for guidance regarding
+how to make pre-release builds available with the CI infrastructure on
+conda-forge. This document outlines how those builds should be labeled on
+anaconda.org so that the more adventurous users can try out these pre-release
+packages.
+
+## Motivation
+
+Pypi allows package maintainers to upload "prerelease" builds which users can
+then install via
+
+```
+pip install PACKAGE_NAME --pre
+```
+
+The same semantics are slightly more cumbersome within the `conda` ecosystem but
+are still manageable.  The analogous command with `conda` is (for example):
+
+```
+conda install -c conda-forge/label/rc_matplotlib matplotlib
```

Comment:
What are anaconda.org's restrictions on valid labels?

1. Are all valid package name characters valid in labels?
2. Is `/` valid, too?

We have to get confirmation for 1. from Anaconda before giving our blessing.

If 2. is possible too, I would like to propose `rc/matplotlib` as the label. It would be more readable and avoid any future ambiguities.

Reply:
+1 to `conda-forge/label/rc/matplotlib` if that is possible, rather than mangling the name of the package.

Reply:
I'm worried that this might imply that there is an `rc` channel with a `matplotlib` channel inside it, which is what we want to avoid if possible. @msarahan @soapy1 thoughts?

Reply:
> I'm worried that this might imply that there is an `rc` channel with a `matplotlib` channel inside it

I don't share that worry. However, anaconda.org does not seem to like it:
You can create such label, but you get a 404 for the `repodata.json`. See
https://anaconda.org/mbargull/_label-test
https://conda.anaconda.org/mbargull/label/rc/_label-test/noarch/repodata.json

I tried out some other characters that the anaconda.org web GUI spits out as valid. A single `:` seems to work, `::` not.

https://conda.anaconda.org/mbargull/label/rc:_label-test/noarch/repodata.json
https://conda.anaconda.org/mbargull/label/rc::_label-test/noarch/repodata.json
```sh
> CONDA_DEFAULT_CHANNELS= conda search 'mbargull/label/rc::_label-test::_label-test'
Loading channels: done
No match found for: mbargull/label/rc::_label-test::_label-test. Search: mbargull/label/rc::_label-test::*_label-test*

PackagesNotFoundError: The following packages are missing from the target environment:
  - mbargull/label/rc::_label-test::_label-test


> CONDA_DEFAULT_CHANNELS= conda search 'mbargull/label/rc:_label-test::_label-test'
Loading channels: done
# Name                       Version           Build  Channel             
_label-test                    0.1.0               0  mbargull/label/rc:_label-test
```

This would mean `rc:matplotlib` or `matplotlib:rc` would be possible. IDK if this would confuse users of the `channel::package` notation, though. And I don't know if this would be officially supported.

We need someone from Anaconda to comment on this. My vote would be to make anaconda.org support `/` in labels, e.g., `rc/matplotlib` or `matplotlib/rc`.